src/commands/deploy.ts

```diff
@@ -16,140 +13,65 @@ export default class Deploy extends Command {
-
-            if (authResult.status === 'error' || !authResult.data) {
-                throw new Error('Invalid username and password or missing 2FA code (if active)');
-            }
-        } else {
-            const verificationResult = await fetch(this.normalizeUrl(flags.url, '/api/v1/me'), {
-                method: 'GET',
-                headers: {
-                    'Content-Type': 'application/json',
-                    'X-Auth-Token': flags.token,
-                    'X-User-Id': flags.userid,
+            },
+            {
+                title: 'Getting Server Info',
+                task: async ()  => {
+                    try {
+                        serverInfo = await getServerInfo(fd);
+                    } catch (e) {
+                        throw new Error(e.message);
+                    }
                 },
-            }).then((res: Response) => res.json());
-
-            if (!verificationResult.success) {
-                throw new Error('Invalid API token');
-            }
-
-            authResult = { data: { authToken: flags.token, userId: flags.userid } };
-        }
-
-        let endpoint = '/api/apps';
-        if (flags.update) {
-            endpoint += `/${fd.info.id}`;
-        }
-
-        const deployResult = await fetch(this.normalizeUrl(flags.url, endpoint), {
-            method: 'POST',
-            headers: {
-                'X-Auth-Token': authResult.data.authToken,
-                'X-User-Id': authResult.data.userId,
             },
-            body: data,
-        }).then((res: Response) => res.json());
-
-        if (deployResult.status === 'error') {
-            throw new Error(`Unknown error occurred while deploying ${JSON.stringify(deployResult)}`);
-        } else if (!deployResult.success) {
-            if (deployResult.status === 'compiler_error') {
-                throw new Error(`Deployment compiler errors: \n${ JSON.stringify(deployResult.messages, null, 2) }`);
-            }
-            throw new Error(`Deployment error: ${ deployResult }`);
-        }
-    }
-
-    // expects the `path` to start with the /
-    private normalizeUrl(url: string, path: string): string {
-        return url.replace(/\/$/, '') + path;
+            {
+                title: 'Packaging',
+                task: async (ctx, task) => {
+                    try {
+                        ctx.zipName = await packageAndZip(this, fd);
+                    } catch (e) {
+                        throw new Error(e.message);
+                    }
+                },
+            },
+            {
+                title: 'Deploying',
+                task: async (ctx, task) => {
+                    try {
+                        await uploadApp({...flags, ...serverInfo}, fd, ctx.zipName);
+                    } catch (e) {
+                        throw new Error(e.message);
+                    }
+                },
+            },
+        ]);
+        tasks.run().catch((e) => {
+            return;
```

Comment:
Please avoid no action in the catch block, at least log the error to the stdout.

Reply:
This is done to avoid repeated showing of the same errors, because if any  command in the tasks array fails it would lead to throw of the error which will be shown once by listr itself and then it would bubble up to the catch block here. As it is already displayed I have not tried to log it again here

Reply:
alright, it makes sense.